onmt/translate/translator.py

```diff
@@ -517,10 +541,90 @@ def _translate_random_sampling(
                     lambda state, dim: state.index_select(dim, select_indices))
 
         results["scores"] = random_sampler.scores
-        results["predictions"] = random_sampler.predictions
+        results["predictions"] = random_sampler.predictions  # (batch, )
         results["attention"] = random_sampler.attention
+        if self.report_align:
+            results["alignment"] = self._align_forward(
```

Comment:
Aren't we supposed to use the already performed forward to get alignments? I reckon this second pass would hurt inference time a lot, wouldn't it?

Reply:
I think this is necessary for full_context_alignment, as in decoder time we generate translation regressively, we can only perform future aware decode after having the prediction.
And this will hurt inference a little bit rather than a lot, as most of the time is used during beam search, this is just a forward pass to get the inner hidden states(attention).